parser/main.py

```diff
@@ -188,215 +194,125 @@ def parse_request():
-        app.logger.info(f"Successfully parsed CAN message with id={can_msg.hex_identifier}({can_msg.identifier}) and placed into queue")
-    except Exception:
-        app.logger.warn(
-            f"Unable to extract measurements for CAN message with id={can_msg.hex_identifier}({can_msg.identifier})")
-        app.logger.warn(str(extract_measurements))
-        return {
-            "result": "PARSE_FAIL",
-            "measurements": [],
-            "id": can_msg.identifier
-        }
-
-    # try putting the extracted measurements in the queue for Grafana streaming
-    try:
-        stream_queue.put(extracted_measurements, block=False)
-    except queue.Full:
-        app.logger.warn(
-            "Stream queue full. Unable to add measurements to stream queue!"
-        )
-
-    # try writing the measurements extracted
-    for measurement in extracted_measurements:
-        name = measurement.name
-        source = measurement.source
-        m_class = measurement.m_class
-        value = measurement.value
-
-        point = influxdb_client.Point(source).tag("car", CAR_NAME).tag(
-            "class", m_class).field(name, value)
-
-        # write to InfluxDB
-        try:
-            write_api.write(bucket=INFLUX_DEBUG_BUCKET, org=INFLUX_ORG, record=point)
-            app.logger.info(
-                f"Wrote '{name}' measurement to url={INFLUX_URL}, org={INFLUX_ORG}, bucket={INFLUX_DEBUG_BUCKET}!")
-        except Exception:
-            app.logger.warning("Unable to write measurement to InfluxDB!")
-            return {
-                "result": "INFLUX_WRITE_FAIL",
-                "measurements": extracted_measurements,
-                "id": can_msg.identifier
-            }
+    app.logger.info(f"Received a {message.type} message. ID = {id=}")
 
     return {
         "result": "OK",
-        "measurements": extracted_measurements,
-        "id": can_msg.identifier
+        "message": message.data["display_data"],
+        "id": id,
+        "type": type
     }
 
 
+@app.post(f"{API_PREFIX}/parse/write/debug")
+@auth.login_required
+def parse_and_write_request():
+    return parse_and_write_request_bucket("_test")
+
 @app.post(f"{API_PREFIX}/parse/write/production")
 @auth.login_required
 def parse_and_write_request_to_prod():
-    """
-    Parses incoming request, writes the parsed measurements to InfluxDB production bucket,
-    and sends back parsed measurements back to client.
-    """
-    parse_request = flask.request.json
-    id: str = parse_request["id"]
-    data: str = parse_request["data"]
-    # TODO: use timestamp when writing to Influx
-    timestamp: str = parse_request["timestamp"]
-    data_length: str = parse_request["data_length"]
-
-    app.logger.info(f"Received message: {id=}, {data=}")
+    return parse_and_write_request_bucket("_prod")
 
-    # TODO: add validation for received JSON object
+"""
+Parses incoming request, writes the parsed measurements to InfluxDB bucket (debug or production),
```

Comment:
This docstring isn't valid "there is no "production bucket"". You could say "production buckets".

Reply:
@AarjavJain101 I would suggest going through any existing comments/docstrings in link_telemetry.py and main.py for the parser and updating them if required.

Reply:
Thanks I have changed the comments and docstrings now